hail/python/hailtop/batch/batch_pool_executor.py

```diff
@@ -478,21 +479,31 @@ async def async_cancel(self):
         ``True`` is returned if the job is cancelled. ``False`` is returned if
         the job has already completed.
         """
+        async def cancel_batch():
+            await self.batch.cancel()
+            self._cancelled_batch = True
+
         if self.fetch_coro.cancelled():
+            status = await self.batch.status()
+            if status['state'] == 'running':
+                await cancel_batch()
```

Comment:
Isn't cancel idempotent and doesn't care if the batch is finished or not? We shouldn't add a round trip to check the status if unnecessary.